pkg/logs/internal/launchers/file/provider/file_provider.go

```diff
@@ -0,0 +1,281 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package fileprovider
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+
+	tailer "github.com/DataDog/datadog-agent/pkg/logs/internal/tailers/file"
+	"github.com/DataDog/datadog-agent/pkg/logs/sources"
+	"github.com/DataDog/datadog-agent/pkg/logs/status"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/pkg/logs/config"
+)
+
+// OpenFilesLimitWarningType is the key of the message generated when too many
+// files are tailed
+const openFilesLimitWarningType = "open_files_limit_warning"
+
+// wildcardOrdering controls what ordering is applied to wildcard files
+type wildcardOrdering string
```

Comment:
Typically such enumeration constants would be numeric, using iota. Is there a reason to make it a string? (Ditto for selectionStrategy)

Reply:
Also, are these feature flags that we'd like to eventually do away with, or permanent parts of the config surface?

Reply:
No particular reason, still learning how to write idiomatic go. I'll switch this to a enumeration constant.

Reply:
@djmitche missed your question, but the config option that is exposed to users is intended to be permanent.

These bits are the flags that `file_provider` exposes to configure the "mode", and per Vickenty's advice I've consolidated them into a single flag.